### PR TITLE
Polymorphic pager

### PR DESCRIPTION
**Pull request by revonateB0T**

### CODE REVIEW

File: app/src/main/kotlin/com/hippo/ehviewer/ui/reader/GalleryPager.kt [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
+                    item(items[index])
+                }
+            }
+        }
+        WEBTOON, CONTINUOUS_VERTICAL -> {
+            LazyColumn(
+                modifier = modifier,
+                state = lazyListState,
+                verticalArrangement = Arrangement.spacedBy(if (type == WEBTOON) 15.dp else 0.dp),
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/reader/GalleryPager.kt:62]: `CONTINUOUS_VERTICAL` is the one has gaps.

**Review by FooIbar — APPROVED** [review]

File: app/src/main/kotlin/com/hippo/ehviewer/ui/reader/ReaderScreen.kt [hunk 2]
```diff
@@ -68,25 +72,38 @@ fun ReaderScreen(info: BaseGalleryInfo, page: Int = -1) {
...
                 onClickSettings = { },
                 modifier = Modifier.zIndex(1f),
             )
-            LazyColumn(modifier = Modifier.fillMaxSize().zoomable(zoomableState), state = lazyListState) {
-                items(pageLoader.mPages) { page ->
+            CompositionLocalProvider(LocalTextStyle provides MaterialTheme.typography.bodySmall) {
+                PageIndicatorText(
+                    currentPage = sync.sliderValue,
+                    totalPages = pageLoader.size,
+                    modifier = Modifier.align(Alignment.BottomCenter).zIndex(0.5f),
```
Comment (FooIbar) [inline app/src/main/kotlin/com/hippo/ehviewer/ui/reader/ReaderScreen.kt:98]: Navigation bar padding, same for ReaderAppBars.

File: app/src/main/kotlin/com/hippo/ehviewer/ui/reader/GalleryPager.kt (hunk 1 above)
Comment (revonateB0T) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/reader/GalleryPager.kt]: Done

File: app/src/main/kotlin/com/hippo/ehviewer/ui/reader/ReaderScreen.kt (hunk 2 above)
Comment (revonateB0T) [reply app/src/main/kotlin/com/hippo/ehviewer/ui/reader/ReaderScreen.kt]: Done, cannot do it for ReaderAppBars now, must control systemBar visibility & color first

**Pull request merged**